test_runner/performance/test_perf_pgbench.py

```diff
@@ -0,0 +1,76 @@
+import os
+import timeit
+from contextlib import closing
+from fixtures.zenith_fixtures import PostgresFactory, ZenithPageserver
+import psycopg2.extras
+
+pytest_plugins = ("fixtures.zenith_fixtures", "fixtures.benchmark_fixture")
+
+def get_timeline_size(repo_dir, tenantid, timelineid):
+    path = "{}/tenants/{}/timelines/{}".format(repo_dir, tenantid, timelineid)
+
+    totalbytes = 0
+    for root, dirs, files in os.walk(path):
+        print(root, "consumes", end=" ")
+        for name in files:
+            totalbytes += os.path.getsize(os.path.join(root, name))
+
+        if 'wal' in dirs:
+            dirs.remove('wal')  # don't visit 'wal' subdirectory
+
+    return totalbytes
+
+#
+# Run a very short pgbench test.
+#
+# Collects three metrics:
+#
+# 1. Time to initialize the pgbench database (pgbench -s5 -i)
+# 2. Time to run 5000 pgbench transactions
+# 3. Disk space used
+#
+def test_pgbench(postgres: PostgresFactory, pageserver: ZenithPageserver, pg_bin, zenith_cli, zenbenchmark):
+    # Create a branch for us
+    zenith_cli.run(["branch", "test_pgbench_perf", "empty"])
+
+    pg = postgres.create_start('test_pgbench_perf')
+    print("postgres is running on 'test_pgbench_perf' branch")
+
+    # Open a connection directly to the page server that we'll use to force
+    # flushing the layers to disk
+    psconn = pageserver.connect();
+    pscur = psconn.cursor(cursor_factory = psycopg2.extras.DictCursor)
+
+    # Get the timeline ID of our branch. We need it for the 'do_gc' command
+    with closing(pg.connect()) as conn:
+        with conn.cursor() as cur:
+            cur.execute("SHOW zenith.zenith_timeline")
+            timeline = cur.fetchone()[0]
+
+    connstr = pg.connstr()
+
+    start = timeit.default_timer()
+
+    pg_bin.run_capture(['pgbench', '-s5', '-i', connstr])
+
+    # Flush the layers from memory to disk. The time to do that is included in the
+    # reported init time.
+    pscur.execute(f"do_gc {pageserver.initial_tenant} {timeline} 0")
+    end = timeit.default_timer()
+
+    zenbenchmark.record('init', end - start, 's')
```

Comment:
Just as an idea, duration recording can be implemented with context managers, like so:
```python
with zenbenchmark.record_duration('init'):
    do_some_work()
```

Reply:
Good idea. I implemented that.

Would be nice to have a similar mechanism for getting the disk space usage. It's a bit more complicated as you need to connect to the page server to flush out the data that's still in-memory, etc. I didn't implement that yet.